WalletWasabi.Fluent/ViewModels/Wallets/Home/History/Details/TransactionDetailsViewModel.cs

```diff
@@ -98,23 +97,16 @@ protected override void OnNavigatedTo(bool isInHistory, CompositeDisposable disp
 		base.OnNavigatedTo(isInHistory, disposables);
 
 		_wallet.Transactions.TransactionProcessed
-							.DoAsync(async _ => await UpdateCurrentTransactionAsync())
+							.Do(_ => UpdateCurrentTransaction())
 							.Subscribe()
 							.DisposeWith(disposables);
 	}
 
-	private async Task UpdateCurrentTransactionAsync()
+	private void UpdateCurrentTransaction()
 	{
-		if (TransactionId is null)
-		{
-			return;
-		}
-
-		var currentTransaction = await _wallet.Transactions.GetById(TransactionId);
-
-		if (currentTransaction is { })
+		if (_wallet.Transactions.TryGetById(TransactionId, out var transactionSummary))
 		{
-			UpdateValues(currentTransaction);
+			UpdateValues(transactionSummary!);
```

Comment:
Can the source gen detect in the implementation of `TryGetById()` the `[NotNullWhen(true)]` part, and also add it to the interface?

@ichthus1604 @SuperJMN

Reply:
Addition:

It wouldn't make sense to add the attribute to an interface as it cannot be guaranteed that all the implementations will add it as well.
But, since we have source-generated interfaces, if all the implementations added the attribute, the interface could have it too. With this the null warning wouldn't have to be eliminated with `!`, which is not really nice